ethereum/spec/src/test/java/tech/pegasys/teku/spec/schemas/registry/BaseSchemaProviderTest.java

```diff
@@ -13,80 +13,175 @@
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaBellatrix");
+
+    assertThat(provider.getSupportedMilestones())
+        .containsAll(SpecMilestone.getAllMilestonesFrom(ALTAIR));
+  }
+
+  @Test
+  void shouldSupportContinuousConstantWithUntil() {
+    final SchemaProvider<?> provider =
+        providerBuilder(STRING_SCHEMA_ID)
+            .constant(true)
+            .withCreator(PHASE0, (r, c) -> "TestSchemaPhase0")
+            .withCreator(BELLATRIX, (r, c) -> "TestSchemaBellatrix")
+            .until(CAPELLA)
+            .build();
+
     assertEquals(PHASE0, provider.getEffectiveMilestone(PHASE0));
     assertEquals(PHASE0, provider.getEffectiveMilestone(ALTAIR));
     assertEquals(BELLATRIX, provider.getEffectiveMilestone(BELLATRIX));
+    assertEquals(BELLATRIX, provider.getEffectiveMilestone(CAPELLA));
+
+    when(mockRegistry.getMilestone()).thenReturn(PHASE0);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaPhase0");
+
+    when(mockRegistry.getMilestone()).thenReturn(ALTAIR);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaPhase0");
+
+    when(mockRegistry.getMilestone()).thenReturn(BELLATRIX);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaBellatrix");
+
+    when(mockRegistry.getMilestone()).thenReturn(CAPELLA);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaBellatrix");
+
+    assertThat(provider.getSupportedMilestones())
+        .containsExactly(PHASE0, ALTAIR, BELLATRIX, CAPELLA);
   }
 
   @Test
-  void shouldGetSchema() {
+  void shouldSupportContinuousDefaultVariable() {
+    final SchemaProvider<?> provider =
+        providerBuilder(STRING_SCHEMA_ID)
+            .withCreator(PHASE0, (r, c) -> "TestSchema" + r.getMilestone())
+            .until(CAPELLA)
+            .build();
+
+    // variable has effective milestone always equal to the milestone
+    SpecMilestone.getMilestonesUpTo(CAPELLA)
+        .forEach(milestone -> assertEquals(milestone, provider.getEffectiveMilestone(milestone)));
+
     when(mockRegistry.getMilestone()).thenReturn(PHASE0);
-    String result = provider.getSchema(mockRegistry);
-    assertEquals("TestSchema", result);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaPHASE0");
+
+    when(mockRegistry.getMilestone()).thenReturn(ALTAIR);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaALTAIR");
+
+    when(mockRegistry.getMilestone()).thenReturn(BELLATRIX);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaBELLATRIX");
+
+    when(mockRegistry.getMilestone()).thenReturn(CAPELLA);
+    assertEquals(provider.getSchema(mockRegistry), "TestSchemaCAPELLA");
+
+    assertThat(provider.getSupportedMilestones())
+        .containsExactly(PHASE0, ALTAIR, BELLATRIX, CAPELLA);
   }
 
   @Test
-  void shouldGetNonOverlappingVersionMappings() {
-    provider.addMilestoneMapping(PHASE0, ALTAIR);
-    provider.addMilestoneMapping(BELLATRIX, CAPELLA);
+  void shouldAllowSingleMilestone() {
```

Comment:
should update test name. Probably is covered by `shouldThrowWhenAskingForAnUnsupportedMilestone`